src/pages/docs/guides/chat/build-livestream.mdx

```diff
@@ -0,0 +1,354 @@
+  * The average message rate is the more important metric to consider when designing your chat, as it represents the overall experience you want to provide.
+* What balance do you want to strike between realtime delivery and cost efficiency?
+  * Batching messages naturally introduces a delay in the time it takes for a message to be delivered to the user, or acknowledged by the server.
+  * Larger batching intervals mean less frequent delivery of messages, but also means that fewer messages in total are sent out (and thus lower costs).
+  * For a large room chat scenario, we recommend a batching interval of **no more than 1 second**, which keeps the chat responsive but can still [significantly reduce costs](#server-side-batching).
+  * For smaller groups and 1:1 chats, you may find that batching does not make a material difference to costs, but does affect the latency of the chat. In the rooms handling those chats, you may choose to not use batching.
+
+## Authentication
+
+Authentication in livestream chat is a design decision. In some cases, you'll want to tie participants to their identity in your application. In others, anyone can join and participate. Ably Chat supports both of these scenarios - you don't need to go through the hassle of setting up users in Ably and trying to keep them in sync with your application. Every client has a `clientId`. If you want to allow random users, just use a random string. If you want to tie it to your application, then use the UUID of the user or some other well-known identifier.
+
+<Code>
+```javascript
+const jwt = require("jsonwebtoken");
+
+const header = {
+    "typ": "JWT",
+    "alg": "HS256",
+    "kid": "{{ API_KEY_NAME }}"
+}
+const currentTime = Math.round(Date.now() / 1000);
+
+// The clientID could be the users UUID in your application,
+// or a random string if you want to make users anonymous.
+//
+// The capabilities here allow the holder of this token to publish and subscribe to messages,
+// typing indicators and room reactions in the chat room called "foo".
+
+const claims = {
+    "iat": currentTime, /* current time in seconds */
+    "exp": currentTime + 14400, /* time of expiration in seconds */
+    "x-ably-capability": "{\"[chat]foo\":[\"publish\", \"subscribe\"]}",
+    "x-ably-clientId": "your-client-id",
+}
+
+ablyJwt = jwt.sign(
+    claims,
+    "{{ API_KEY_SECRET }}",
+    { header: header }
+)
+
+console.log('JWT is: ' + ablyJwt);
+```
+</Code>
+
+How you authenticate is also key. To balance security and experience, you want short-lived tokens that can be easily revoked if a users is misbehaving or needs their permissions changed, but also avoids the overhead of issuing new tokens every 2 minutes. **In production apps, you should not use API keys for client-side authentication**. You can use them server-side, but as they are long-lived they are much more difficult to revoke in a hurry.
```

Comment:
Why are API keys more difficult to revoke?  You can remove them immediately and that makes them invalid there and then. With tokens, unless you use the revoke fature, the lag can be the renewal time. This statement does not feel correct. The point is client-side tokens are for untrusted clients, and can be revoked or blocked from renewal. Too frequently renewing tokens creates second order load problems with the token issuer, and that needs to be considere

Reply:
Reworded